runtime/src/commonMain/kotlin/pbandk/json/JsonWireMarshaller.kt

```diff
@@ -0,0 +1,183 @@
+        FieldDescriptor.Type.Primitive.Int32 -> writeNumber32(value as Int)
+        FieldDescriptor.Type.Primitive.Fixed64 -> writeNumber64(value as Long)
+        FieldDescriptor.Type.Primitive.Fixed32 -> writeNumber32(value as Int)
+        FieldDescriptor.Type.Primitive.Bool -> writeBool(value as Boolean)
+        FieldDescriptor.Type.Primitive.String -> writeString(value as String)
+        FieldDescriptor.Type.Primitive.Bytes -> writeBytes(value as ByteArr)
+        FieldDescriptor.Type.Primitive.UInt32 -> writeNumber32(value as Int)
+        FieldDescriptor.Type.Primitive.SFixed32 -> writeNumber32(value as Int)
+        FieldDescriptor.Type.Primitive.SFixed64 -> writeNumber64(value as Long)
+        FieldDescriptor.Type.Primitive.SInt32 -> writeNumber32(value as Int)
+        FieldDescriptor.Type.Primitive.SInt64 -> writeNumber64(value as Long)
+        is FieldDescriptor.Type.Message -> when (type.messageCompanion) {
+            // Wrapper types use the same JSON representation as the wrapped value
+            // https://developers.google.com/protocol-buffers/docs/proto3#json
+            is DoubleValue.Companion -> writeDouble(value as Double)
+            is FloatValue.Companion -> writeFloat(value as Float)
+            is Int64Value.Companion -> writeNumber64(value as Long)
+            is UInt64Value.Companion -> writeNumber64(value as Long)
+            is Int32Value.Companion -> writeNumber32(value as Int)
+            is UInt32Value.Companion -> writeNumber32(value as Int)
+            is BoolValue.Companion -> writeBool(value as Boolean)
+            is StringValue.Companion -> writeString(value as String)
+            is BytesValue.Companion -> writeBytes(value as ByteArr)
+            // Other well-known types with special JSON encoding
+            is Timestamp.Companion -> writeString(
+                Util.timestampToString(
+                    value as Timestamp
+                )
+            )
+            // All other message types
+            else -> writeMessage(value as Message)
+        }
+        is FieldDescriptor.Type.Enum -> writeEnum(value as Message.Enum)
+        is FieldDescriptor.Type.Repeated -> writeRepeated(value as List<*>, type.valueType)
+        is FieldDescriptor.Type.Map -> writeMap(value as Map<*, *>, type.keyType, type.valueType)
+    }
+
+    fun writeNumber32(value: Int): JsonElement =
+        JsonLiteral(value)
+
+    fun writeNumber64(value: Long): JsonElement =
+        JsonLiteral(value.toString())
+
+//    fun writeInt32(value: Int): JsonElement = JsonLiteral(value)
+//
+//    fun writeInt64(value: Long): JsonElement = JsonLiteral(value.toString())
+//
+//    fun writeUInt32(value: Int): JsonElement = JsonLiteral(value)
+//
+//    fun writeUInt64(value: Long): JsonElement = JsonLiteral(value.toString())
+//
+//    fun writeSInt32(value: Int): JsonElement = JsonLiteral(value)
+//
+//    fun writeSInt64(value: Long): JsonElement = JsonLiteral(value.toString())
```

Comment:
Why commented out?

Reply:
This is some of the cleanup I mentioned I still need to do. These are currently all going through the generic `writeNumber32` and `writeNumber64` methods instead, since JSON uses the same serialization for all 32-bit and all 64-bit number types. Though I'm considering bringing these methods back and having `JsonValueMarshaller` extend from the `ValueMarshaller` interface (it doesn't currently), which I think will enable some useful behavior in the future.